### DISCUSSION TITLE
patch-package fails on initial deploy, but not on redeploy

### DISCUSSION THREAD

**Discussion by bigbizze** [vercel/community#2154] [Bug Report]
next=13.1.5
patch-package=6.5.1

Prior to this morning, we had no issues with patch-package being applied in the package.json postinstall script in our vercel deployment. However, now when we update our repository and vercel does a build of our application, we get:

```
This error was caused because patch-package cannot apply the following patch file:
    patches/@azure+ai-form-recognizer+4.0.0.patch
  Try removing node_modules and trying again. If that doesn't work, maybe there was
  an accidental change made to the patch file? Try recreating it by manually
  editing the appropriate files and running:
  
    patch-package @azure/ai-form-recognizer
  
  If that doesn't work, then it's a bug in patch-package, so please submit a bug
  report. Thanks!
    https://github.com/ds300/patch-package/issues
    
---
patch-package finished with 1 error(s).
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
Error: Command "yarn install" exited with 1
Deployment completed
BUILD_UTILS_SPAWN_1: Command "yarn install" exited with 1
```

And the build fails. What makes this strange, and why I am making a report here rather elsewhere, is that when I click "Redeploy" on the build, and build without the existing build cache, it builds correctly and without issues. Additionally, we had no such issues prior to today with vercel builds with the same application and even for the same commit.

**Comment by federicogomez-stowlog** [comment]
Just started having the exact same issue a few days ago, and indeed when manually redeploying everything is ok

```

This error was caused because patch-package cannot apply the following patch file:
--
04:05:01.663 |  
04:05:01.663 | patches/react-native-calendars+1.1297.0.patch
04:05:01.663 |  
04:05:01.663 | Try removing node_modules and trying again. If that doesn't work, maybe there was
04:05:01.663 | an accidental change made to the patch file? Try recreating it by manually
04:05:01.663 | editing the appropriate files and running:
04:05:01.663 |  
04:05:01.663 | patch-package react-native-calendars
04:05:01.663 |  
04:05:01.663 | If that doesn't work, then it's a bug in patch-package, so please submit a bug
04:05:01.663 | report. Thanks!
04:05:01.663 |  
04:05:01.664 | https://github.com/ds300/patch-package/issues
04:05:01.664 |  
04:05:01.664 |  
04:05:01.664 | ---
04:05:01.664 | patch-package finished with 1 error(s).
04:05:01.678 | error Command failed with exit code 1.
04:05:01.678 | info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
04:05:01.712 | Error: Command "yarn install" exited with 1

```

**Comment by swarnava** [comment]
Hi,

Can you reproduce this constantly? Could you share a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) repository that can be shared publicly as a git repository so that we can take a look?

**Comment by bombillazo** [comment]
This is happening to us as well, having to redeploy with cache skipping.

**Comment by robertontiu** [comment]
Any news on this? Literally happening on every single PR and having to manually redeploy each one without bundle cache is getting a bit ridiculous.

**Comment by tempppabx1** [comment]
The same problem persists, what's confusing is that not every patched module will appear. At one point, I event considered using other complex patching methods.
However, When I manually redeployed and the build was sucessful, I realized that is problem might be related to caching. So, the temporary solution was to add the enviroment variable `VERCEL_FORCE_NO_BUILD_CACHE=1`, which prevents Vercel from using existing cache during automatic builds.

**Comment by PascalPixel** [comment]
If Vercel diffed the `patches/` folder for changes between deployments it would know not to use the cache, for now, adding the `VERCEL_FORCE_NO_BUILD_CACHE=1` to env vars in project settings fixes it as @tempppabx1 suggested.

**Reply by PascalPixel** [reply]
My guess is that Vercel internally has moved to `pnpm`.

Since `pnpm` lists its own patches in `package.json`, someone optimized the deployment scripts to only look at changes in `package.json` for when to rebuild the cache, as opposed to also looking at `/patches` itself before.

Makes sense because at scale this would probably save them a lot of time deploying all projects without scanning this folder for changes, and there aren't that many people that use `patch-package`.

**Comment by dcapitator** [comment]
having this issue too. I have to redeploy manually skipping cache
